org.civicrm.reqangular/src/common/services/api.js

```diff
@@ -121,9 +122,13 @@ define([
             action: action
           }
         }).then(function (response) {
-
-          return response.data;
+          if(!!response.data.is_error) {
+            deferred.reject(response.data.error_message);
+          }
+          deferred.resolve(response.data);
         });
+
+        return deferred.promise;
       },
```

Comment:
It is not done intentionally, because the catch will be caught by the actual controller/service(example: `RequestCtrl`). The error automatically gets caught by the first `catch` in the promise chain.